### PR TITLE
VM interface

### PR DESCRIPTION
**Pull request by adlrocha**
*I am splitting [this PR](github.com/libp2p/go-smart-record/pull/8/) into several ones to accommodate for the new VM changes, and ease their review.*

# Goal
This PR introduces the VM interface. The interface is used for the libp2p protocol integration.

### CODE REVIEW

**Comment by adlrocha** [comment]
//cc @barath (I can't add you as a reviewer. I guess it is because you don't have the right permissions).

File: vm/vm.go [hunk 1]
```diff
@@ -2,12 +2,60 @@
...
+	Value RecordValue // Root stores peers dictionaries in that key.
+}
+
 // Machine captures the public interface of a smart record virtual machine.
 type Machine interface {
-	Update(ir.Dict) error
-	Query(ir.Dict) error
-	Get() ir.Dict
+	Update(writer peer.ID, k string, update ir.Dict) error // Updates the dictionary in the writer's private space.
+	Get(k string) Record                                   // Get the full Record in a key
```
Comment (petar) [inline vm/vm.go:23]: This can just return RecordValue. This way you don't need to define the type Record. Simplification.

**Review by petar — APPROVED** [review]
LGTM

File: vm/vm.go (hunk 1 above)
Comment (petar) [reply vm/vm.go:23]: BY THE WAY, don't worry about changing this. You can leave Record as is for now. Just a style comment for the future.

Comment (petar) [reply vm/vm.go:23]: The point being: The returned values should generally not repeat the input. The caller already knows the input.

Comment (adlrocha) [reply vm/vm.go]: No, you are right. Too much redundancy. I rather change it now. Thanks!

**Pull request merged**